packages/std/src/math/decimal256.rs

```diff
@@ -0,0 +1,984 @@
+use schemars::JsonSchema;
+use serde::{de, ser, Deserialize, Deserializer, Serialize};
+use std::fmt::{self, Write};
+use std::ops;
+use std::str::FromStr;
+
+use crate::errors::StdError;
+
+use super::Fraction;
+use super::Isqrt;
+use super::Uint256;
+
+/// A fixed-point decimal value with 36 fractional digits, i.e. Decimal256(1_000_000_000_000_000_000) == 1.0
+///
+/// The greatest possible value that can be represented is
+/// 115792089237316195423570985008687907853269.984665640564039457584007913129639935
+/// (which is (2^256 - 1) / 10^36)
+#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, JsonSchema)]
+pub struct Decimal256(#[schemars(with = "String")] Uint256);
+
+impl Decimal256 {
+    fn decimal_fractional() -> Uint256 {
+        // 1*10**36
+        Uint256::from(1_000_000_000_000_000_000_000_000_000_000_000_000u128)
+    }
+
+    fn decimal_fractional_squared() -> Uint256 {
+        // 1*10**72
+        Uint256::new([
+            0, 0, 144, 228, 15, 190, 234, 29, 58, 74, 188, 137, 85, 233, 70, 254, 49, 205, 207,
+            102, 246, 52, 225, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        ])
+    }
```

Comment:
It would be cool to have some const constructors for U256. But they don't seem to exist. Could you ask the uint authors if they can/want to provide them?

Reply:
https://github.com/paritytech/parity-common/issues/585

Reply:
Thanks. Given the response there, does this mean we can implement `Uint256::from_be_bytes` and `Uint256::from_le_bytes` as const functions by bushing bytes around?

Reply:
Might be tricky to do that when we're only limited to `const` function calls. What's the priority on this?

Reply:
Good call. Not important since we can change this any time.

Reply:
See https://github.com/CosmWasm/cosmwasm/pull/1071